services/webhook/notifier.go

```diff
@@ -871,14 +872,22 @@ func (m *webhookNotifier) CreateCommitStatus(ctx context.Context, repo *repo_mod
 		return
 	}
 
+	// as a webhook url, target should be an absolute url. But for internal actions target url
+	// the target url is a url path with no host and port to make it easy to be visited
+	// from multiple hosts. So we need to convert it to an absolute url here.
+	target := status.TargetURL
+	if strings.HasPrefix(target, "/") {
+		target = setting.AppURL + strings.TrimPrefix(target, "/")
+	}
```

Comment:
It could simply use `MakeAbsoluteURL`, then no need to use these tricks.

![image](https://github.com/user-attachments/assets/8d923678-744d-443c-9806-c6effc27166a)

Reply:
8f80089a3c